doc/WebSite/SignalR-AspNetCore-Integration.md

```diff
@@ -0,0 +1,179 @@
+real time notifications to clients (see [notification
+system](/Pages/Documents/Notification-System)). Thus, your users can get
+real time push notifications.
+
+#### Online Clients
+
+ASP.NET Boilerplate provides **IOnlineClientManager** to get information
+about online users (inject IOnlineClientManager and use
+GetByUserIdOrNull, GetAllClients, IsOnline methods for example).
+IOnlineClientManager needs a communication infrastructure to properly
+work. **Abp.Web.SignalR.AspNetCore** package provides the infrastructure. So, you
+can inject and use IOnlineClientManager in any layer of your application
+if SignalR is installed.
+
+### Your SignalR Code
+
+**Abp.Web.SignalR.AspNetCore** package also simplifies your SignalR code. Consider
+that we want to add a Hub to our application:
+
+    public class MyChatHub : Hub, ITransientDependency
+    {
+        public IAbpSession AbpSession { get; set; }
+
+        public ILogger Logger { get; set; }
+
+        public MyChatHub()
+        {
+            AbpSession = NullAbpSession.Instance;
+            Logger = NullLogger.Instance;
+        }
+
+        public async Task SendMessage(string message)
+        {
+            await Clients.All.getMessage(string.Format("User {0}: {1}", AbpSession.UserId, message));
+        }
+
+        public override async Task OnConnectedAsync()
+        {
+            await base.OnConnectedAsync();
+            Logger.Debug("A client connected to MyChatHub: " + Context.ConnectionId);
+        }
+
+        public override async Task OnDisconnectedAsync(Exception exception)
+        {
+            await base.OnDisconnectedAsync(exception);
+            Logger.Debug("A client disconnected from MyChatHub: " + Context.ConnectionId);
+        }
+    }
+
+<!-- -->
+
+    routes.MapHub<MyChatHub>("/myChatHub");
+
+We implemented **ITransientDependency** to simply register our hub to
+[dependency injection](/Pages/Documents/Dependency-Injection) system
+(you can make it singleton based on your needs). We
+[property-injected](/Pages/Documents/Dependency-Injection#property-injection-pattern)
+the [session](/Pages/Documents/Abp-Session) and
+[logger](/Pages/Documents/Logging).
+
+**SendMessage** is a method of our hub that can be used by clients. We
+call **getMessage** function of **all** clients in this method. We can
+use [AbpSession](/Pages/Documents/Abp-Session) to get current user id
+(if user logged in) as done above. We also overrided **OnConnectedAsync** and
+**OnDisconnectedAsync**, which is just for demonstration and not needed here
+actually.
+
+Here is the **client side** JavaScript code to send/receive messages using
+our hub.
+
+    var chatHub = null;
+
+    abp.signalr.startConnection('/myChatHub', function (connection) {
+        chatHub = connection; // Save a reference to the hub
+
+        connection.on('getMessage', function (message) { // Register for incoming messages
+            console.log('received message: ' + message);
+        };
```

Comment:
It seems like there is a missing ) here.

Reply:
Will fix.

Reply:
Thanks, then I will confirm the PR.